### PR TITLE
Denraf/feature/base64decode

### PR DESCRIPTION
**Pull request by denraf**
### Description
Adding extra modifier: base64decode

**Fixes:** https://github.com/IBM/argocd-vault-plugin/issues/232

### Checklist
Please make sure that your PR fulfills the following requirements:
- [X] Reviewed the guidelines for contributing to this repository
- [X] The commit message follows the [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [X] Tests for the changes have been updated
- [ ] Docs have been added / updated
- [ ] Optional. My organization is added to USERS.md.

### Type of Change

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

### Other information
I have a build with these two plugins running in my test environment.
Full path to secret:
`secret: <path:path/to/secrets#encoded_secret| base64decode  | jsonPath {.secret_key} >`

### CODE REVIEW

**Comment by werne2j** [comment]
Hi @denraf thanks for this contribution! Could you please do me a favor and split the two modifiers into separate pull requests? This will help us review and merge them independently. Thanks again!

**Review by jkayani — COMMENTED** [review]
This looks good @denraf - would you mind adding some e2e tests to the `fixtures/` directory? Specifically creating a YAML that uses these modifiers both individually and separately in `fixtures/input/nonempty/` and then adding the resulting output to `fixtures/output/all.yaml`?

**Comment by denraf** [comment]
I've been looking into the e2e tests and was wondering where the data comes from that's being used in the output. Any pointers?

**Comment by werne2j** [comment]
Hi @denraf for e2e tests.. this is where the e2e tests are https://github.com/IBM/argocd-vault-plugin/blob/main/cmd/generate_test.go#L113 and this is the Test Vault we execute the command against https://github.com/IBM/argocd-vault-plugin/blob/a270631026d83a9167cd13c9e72a100e129f3556/pkg/helpers/test_helpers.go#L109 so you should be able to see where we are writing to this test vault where we then reference in yaml files in the `fixtures/input/nonempty/` directory. Hopefully that helps!

**Comment by denraf** [comment]
That was exactly what I was looking for. Thank you for you help!

**Comment by werne2j** [comment]
@denraf Could you do me a favor and squash your commits down to two. One for `base64decode` and one for `yamltojson`?

**Comment by teejaded** [comment]
Since modifiers can pass objects through the pipeline would it make sense to split yamltojson up into separate modifiers?

* fromYaml
* toYaml
* toJson

**Comment by werne2j** [comment]
@teejaded I think that is an interesting idea and seems to make sense. You are right, as we can string together multiple modifiers. @denraf Let's change this PR to be just `base64decode` and then we can work on the other modifiers we want to include in other PRs.

**Comment by denraf** [comment]
@teejaded @werne2j I'm not sure it makes sense to split the modifiers up in the suggested way. The standard for anything vault related is json, so when transforming data, it should end up in json.

I will split the PR up for `base64decode` and the `yamltojson` modifier.

**Comment by werne2j** [comment]
@denraf Can you add documentation to the modifier here https://github.com/IBM/argocd-vault-plugin/blob/main/docs/howitworks.md#modifiers following the same pattern as the other modifiers

**Comment by werne2j** [comment]
Looks like the e2e tests are failing - https://github.com/IBM/argocd-vault-plugin/runs/4301432074?check_suite_focus=true#step:5:11

**Review by werne2j — APPROVED** [review]
Looks good!

File: docs/howitworks.md [hunk 1]
```diff
@@ -263,6 +263,18 @@ Valid examples:
...
 
+##### `base64decode`
+
+The base64decode modifier decodes base64 encoded values into plain-text.
+
+Valid examples:
+
+- `<b64_username | base64decode>`
+
+- `<path:secrets/data/my-db#b64_username | base64encode>`
```
Comment (jkayani) [inline docs/howitworks.md:274]: I think you mean `decode` here :)

**Review by jkayani — APPROVED** [review]

File: docs/howitworks.md (hunk 1 above)
Comment (jkayani) [reply docs/howitworks.md:274]: Feel free to fix here, code looks great otherwise so I left an approval

Comment (denraf) [reply docs/howitworks.md]: Great catch! Just updated.

**Pull request merged**